### PR TITLE
RN: Apply native changes to expo module

### PR DESCRIPTION
**Pull request by bangtoven**
### _How did you test your changes?_

manually 
image

### CODE REVIEW

File: react-native/ios/CoinbaseWalletSDKModule.swift [hunk 1]
```diff
@@ -164,7 +165,7 @@ extension ReturnValue {
 
         switch self {
         case .result(let value):
-            record.result = value
+            record.result = value.rawValue
```
Comment (bangtoven) [inline react-native/ios/CoinbaseWalletSDKModule.swift:168]: this is the only place where it's affected by the JSONString change that i made on ios. (the other place is the Web3JSONRPC helper enums, and it doesn't seem like expo module uses that)

File: react-native/ios/CoinbaseWalletSDKModule.swift [hunk 2]
```diff
@@ -73,11 +73,12 @@ public class CoinbaseWalletSDKModule: Module {
...
             }
 
-            CoinbaseWalletSDK.shared.initiateHandshake(initialActions: actions) { result in
+            CoinbaseWalletSDK.shared.initiateHandshake(initialActions: actions) { result, account in
                 switch result {
                 case .success(let response):
                     let results: [ReturnValueRecord.Dict] = response.content.map { $0.asRecord }
-                    promise.resolve(results)
+                    let accountRecord = account?.asRecord
+                    promise.resolve((results, accountRecord))
```
Comment (bangtoven) [inline react-native/ios/CoinbaseWalletSDKModule.swift:81]: not sure this is valid on react native side. it builds on swift.

File: react-native/ios/CoinbaseWalletSDKExpo.podspec [hunk 3]
```diff
@@ -16,7 +16,7 @@ Pod::Spec.new do |s|
   s.static_framework = true
 
   s.dependency 'ExpoModulesCore'
-  s.dependency 'CoinbaseWalletSDK', '0.1.2'
+  s.dependency 'CoinbaseWalletSDK', '1.0.0' # TODO
```
Comment (bangtoven) [inline react-native/ios/CoinbaseWalletSDKExpo.podspec]: this hasn't been published yet.

File: react-native/android/src/main/java/expo/modules.coinbasewalletsdkexpo/CoinbaseWalletSDKModule.kt [hunk 4]
```diff
@@ -62,7 +62,7 @@ class CoinbaseWalletSDKModule : Module() {
                 )
             }
 
-            sdk.initiateHandshake(handshakeActions) { result ->
+            sdk.initiateHandshake(handshakeActions) { result -> // TODO
```
Comment (bangtoven) [inline react-native/android/src/main/java/expo/modules.coinbasewalletsdkexpo/CoinbaseWalletSDKModule.kt:65]: please apply the kotlin changes on providing account object

File: react-native/ios/CoinbaseWalletSDKModule.swift [hunk 5]
```diff
@@ -173,3 +174,14 @@ extension ReturnValue {
...
+
+extension Account {
+    var asRecord: AccountRecord.Dict {
+        let record = AccountRecord()
+        record.chain = self.chain
+        record.networkId = Int(self.networkId)
+        record.address = self.address
+        
+        return record.toDictionary()
+    }
```
Comment (bangtoven) [inline react-native/ios/CoinbaseWalletSDKModule.swift:186]: just followed your code above. feel free to update.

**Review by bangtoven — COMMENTED** [review]
@vishnumad 
i tried to apply the recent native changes to your expo module.
this at least compiles (swift on xcode / ts via `expo-module build`), but i couldn't test it end to end.

File: react-native/android/src/main/java/expo/modules.coinbasewalletsdkexpo/CoinbaseWalletSDKModule.kt [hunk 6]
```diff
@@ -62,11 +62,16 @@ class CoinbaseWalletSDKModule : Module() {
...
                 result
                     .onSuccess { responses ->
                         val results: List<ReturnValueRecord> = responses.map { it.asRecord }
-                        promise.resolve(results)
+                        // TODO
+                        val accountRecord = AccountRecord()
+                        accountRecord.chain = account.chain
+                        accountRecord.networkId = account.networkId
+                        accountRecord.address = account.address
+                        promise.resolve(Pair(results, accountRecord))
```
Comment (bangtoven) [inline react-native/android/src/main/java/expo/modules.coinbasewalletsdkexpo/CoinbaseWalletSDKModule.kt]: not sure this is correct kotlin syntax. couldn't even try compiling this. 
please apply your kotlin changes.

File: react-native/ios/CoinbaseWalletSDKExpo.podspec (hunk 3 above)
Comment (bangtoven) [reply react-native/ios/CoinbaseWalletSDKExpo.podspec]: https://github.com/coinbase/wallet-mobile-sdk/pull/29

File: react-native/src/WalletMobileSDKEVMProvider.ts [hunk 7]
```diff
@@ -676,13 +650,13 @@ export class WalletMobileSDKEVMProvider
...
+  private async _makeHandshakeRequest(action: Action): Promise<[unknown, Account]> {
     try {
-      const [res] = await initiateHandshake([action]);
-      if (res.errorMessage) {
+      const [[res], account] = await initiateHandshake([action]);
+      if (!res.result || !account) {
         throw this._getProviderError(res);
       }
-      return res;
+      return [JSON.parse(res.result), account];
```
Comment (bangtoven) [inline react-native/src/WalletMobileSDKEVMProvider.ts:659]: with the change on our RN app, returning results with `JSON.stringify` we can parse them back into original type. 
image

File: react-native/src/WalletMobileSDKEVMProvider.ts [hunk 8]
```diff
@@ -409,15 +400,11 @@ export class WalletMobileSDKEVMProvider
...
-        result: res.result,
-      };
-    } else {
-      throw this._getProviderError(res);
-    }
+    return {
+      jsonrpc: "2.0",
+      id: 0,
+      result: res,
+    };
```
Comment (vishnumad) [inline react-native/src/WalletMobileSDKEVMProvider.ts:407]: Don't we need to handle the error cases for when res has errorCode and errorMessage?

File: react-native/android/src/main/java/expo/modules.coinbasewalletsdkexpo/CoinbaseWalletSDKModule.kt [hunk 9]
```diff
@@ -62,11 +62,18 @@ class CoinbaseWalletSDKModule : Module() {
...
                         val results: List<ReturnValueRecord> = responses.map { it.asRecord }
-                        promise.resolve(results)
+                        /* 
+                        TODO
+                        val accountRecord = AccountRecord()
+                        accountRecord.chain = account.chain
+                        accountRecord.networkId = account.networkId
+                        accountRecord.address = account.address
+                        promise.resolve([results, accountRecord])
+                        */
```
Comment (bangtoven) [inline react-native/android/src/main/java/expo/modules.coinbasewalletsdkexpo/CoinbaseWalletSDKModule.kt:77]: we can test this out once new version of android sdk gets published. @AMITGOELNY

Comment (vishnumad) [inline react-native/android/src/main/java/expo/modules.coinbasewalletsdkexpo/CoinbaseWalletSDKModule.kt:77]: Need to resolve promise here
```
promise.resolve(results)
```

File: react-native/src/WalletMobileSDKEVMProvider.ts (hunk 8 above)
Comment (bangtoven) [reply react-native/src/WalletMobileSDKEVMProvider.ts:407]: i don't think that code path would get triggered ever.
on `_makeSDKRequest` it checks if res.result exists anyway. 
if not, it goes into `_getProviderError`.
image
@vishnumad

**Review by vishnumad — APPROVED** [review]

**Pull request merged**